libraries/rush-lib/src/cli/actions/DeployAction.ts

```diff
@@ -144,20 +146,25 @@ export class DeployAction extends BaseRushAction {
 
     let transformPackageJson: ((packageJson: IPackageJson) => IPackageJson) | undefined;
     let pnpmInstallFolder: string | undefined;
+
+    const rushConfigurationProject: RushConfigurationProject | undefined =
+      this.rushConfiguration.getProjectByName(mainProjectName);
+    const subspace: Subspace = rushConfigurationProject?.subspace || this.rushConfiguration.defaultSubspace;
```

Comment:
The `mainProjectName` is expected to exist, and if not, then an error will be reported:

https://github.com/microsoft/rushstack/blob/8ff5b5a6f0b1fbd4cd9fa955618866d3b631ab06/libraries/package-extractor/src/PackageExtractor.ts#L370-L375

There fore `|| this.rushConfiguration.defaultSubspace` is not really correct or meaningful -- it is providing a non-undefined value to avoid reporting the error here.

Reply:
Should I instead add a validation check that the rush configuration project can actually be found using the mainProjectName? Even though mainProjectName is validated to exist, I don't see it being validated as an actual rush project

Reply:
🤔 Actually it is probably better for Rush to catch it here, since it can provide a more informative error message than the `package-extractor` library can.